pkg/pipeline/tenant_steps.go

```diff
@@ -83,3 +84,25 @@ func setGlobalGitRepoURL(obj PipelineObject, data *ExecutionContext) ExecutionRe
 	}
 	return ExecutionResult{}
 }
+
+func applyTemplateFromTenantTemplate(obj PipelineObject, data *ExecutionContext) ExecutionResult {
+	key := types.NamespacedName{Name: "default", Namespace: obj.GetObjectMeta().GetNamespace()}
+	template := &synv1alpha1.TenantTemplate{}
+	if err := data.Client.Get(context.TODO(), key, template); err != nil {
+		// The absence of a template is not an error.
+		// It simply means that there is nothing to do.
+		// TODO Consider to log this case. Where to get a logger from?
+		return ExecutionResult{}
+	}
```

Comment:
We can at least ignore the IsNotFound error. Other errors we can hardly ignore like temporary, e.g. network-related errors that may in a later reconciliation be resolved again.
We might consider Conditions here to report back errors, also as a means for the lieutenant-api to respect failed reconciliations (e.g. if there's no Ready condition, throw an API error etc)